DevoxxClientMobile/src/main/java/com/devoxx/views/SponsorPresenter.java

```diff
@@ -26,100 +26,161 @@
-    }
-    public final void setSponsor(Sponsor value) {
-        sponsorProperty.set(value);
-    }
+    @Inject
+    private Service service;
 
+    private CharmListView<SponsorBadge, String> lvBadges;
+    private String name;
+    private String slug;
 
     public void initialize() {
-        webLaunchFAB = Util.createWebLaunchFAB(() -> getSponsor().getSummary());
-        sponsor.getLayers().add(webLaunchFAB.getLayer());
 
-        sponsor.setOnShowing( event -> {
+        sponsorView.setOnShowing(event -> {
             AppBar appBar = getApp().getAppBar();
-            appBar.setNavIcon(getApp().getNavBackButton());
+            appBar.setNavIcon(getApp().getNavMenuButton());
             appBar.setTitleText(DevoxxView.SPONSOR.getTitle());
+            checkAndLoadView();
+        });
+        
+        sponsorView.setOnHiding(event -> {
+            sponsorView.getLayers().clear();
+        });
+    }
 
-            // give images full width of scene
-            sponsor.getScene().widthProperty().addListener((observable, oldValue, newValue) -> resizeImages());
-            sponsor.getScene().heightProperty().addListener((observable, oldValue, newValue) -> resizeImages());
+    public void setSponsor(String name, String slug) {
+        this.name = name;
+        this.slug = slug;
+        checkAndLoadView();
+    }
 
-            // randomly change image on each showing
-            imageView.setImage(Util.getMediaBackgroundImage());
+    @FXML
+    private void signIn() {
+        final GluonObservableObject<String> passwordObject = service.authenticateSponsor(password.getText());
+        passwordObject.stateProperty().addListener((o, ov, nv) -> {
+            if (nv == ConnectState.SUCCEEDED) {
+                Services.get(SettingsService.class).ifPresent(service -> {
+                    service.store(Sponsor.NAME, name);
+                    service.store(Sponsor.SLUG, slug);
+                });
+                loadAuthenticatedView(name, slug);
+                message.setText("");
+                password.clear();
+            } else if (nv == ConnectState.FAILED) {
+                message.setText("Verification failed");
+            }
         });
     }
 
-    private void resizeImages() {
-        if (sponsor == null || sponsor.getScene() == null) {
-            return;
-        }
-        
-        double newWidth = sponsor.getScene().getWidth();
-        double newHeight = sponsor.getScene().getHeight() - getApp().getAppBar().getHeight(); // Exclude the AppBar
-
-        logo.setFitWidth(newWidth);
-        // Resize logo to a max value. Do not stretch anymore if the width threshold.
-        Util.resizeImageView(logo, newWidth, newHeight / 3.0);
+    private void loadAuthenticatedView(String name, String slug) {
+        final AppBar appBar = getApp().getAppBar();
+        final Button shareButton = getApp().getShareButton();
+        final Button logoutButton = MaterialDesignIcon.POWER_SETTINGS_NEW.button(e -> {
```

Comment:
I find this sign out button a bit confusing. Do we perhaps want to warn the sponsor first, instead of signing off directly?

Reply:
When the user logs out, they log out as the sponsor. Therefore, it takes them back to the Sponsors (sponsor selection) view.